arbnode/batch_poster.go

```diff
@@ -1516,7 +1524,11 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		// don't post anything for now
 		return false, nil
 	}
-
+	// If we are checking the validation, set isWaitingForEspressoValidation in the batch segments and re-poll the function until we are ready to post.
```

Comment:
I'm also a bit worried about this with respect to the escape hatch's correctness. The current implementation of the escape hatch decides if hotshot is live based on its most recent header according to the light client.
I think its possible that we could submit some transactions to espresso, then, if liveness has changed, skip verifying them by the time we decide to it's time to post the batch and then check the validation. I'm not sure how we would easily handle this though.

Reply:
I think that is expected behaviour though? if hotshot is not live, we should not wait for espresso validation. CC @jbearer  correct me if my understanding is incorrect here.

Reply:
I think that's not a problem. If the escape hatch has triggered than all bets are off anyways when it comes to Espresso confirmations. In other words the guarantees around Espresso confirmations are always "assuming Espresso remains secure and live". The escape hatch is just there so that the rollup as a whole will continue running if Espresso goes down, but we don't have to worry about the integrity of espresso confirmations in this case

Reply:
Which is exactly why we prefer for rollups not to enable the escape hatch if possible